src/extension/tab/hook.ts

```diff
@@ -0,0 +1,156 @@
+import { gql, Observable, ApolloClient } from "@apollo/client";
+import { getMainDefinition } from "@apollo/client/utilities";
+import { version as devtoolsVersion } from "../manifest.json";
+import Relay from "../../Relay";
+import { GraphiQLResponse, QueryResult } from '../../types';
+import { 
+  DEVTOOLS_INITIALIZED,
+  CREATE_DEVTOOLS_PANEL,
+  ACTION_HOOK_FIRED, 
+  GRAPHIQL_REQUEST,
+  GRAPHIQL_RESPONSE,
+  REQUEST_DATA,
+  UPDATE,
+} from "../constants";
+
+declare global {
+  type TCache = any;
+
+  interface Window {
+    __APOLLO_CLIENT__: ApolloClient<TCache>
+  }
+}
+
+type Hook = {
+  ApolloClient: ApolloClient<TCache> | undefined;
+  version: string;
+  getQueries: () => void;
+  getMutations: () => void;
+  getCache: () => void;
+}
+
+function initializeHook() {
+  const hook: Hook = {
+    ApolloClient: undefined,
+    version: devtoolsVersion,
+    getQueries: () => {},
+    getMutations: () => {},
+    getCache: () => {},
+  };
+
+  Object.defineProperty(window, '__APOLLO_DEVTOOLS_GLOBAL_HOOK__', {
```

Comment:
Totally optional, but you might want to consider adding `__APOLLO_DEVTOOLS_GLOBAL_HOOK__` to the `Window` interface above (you could then move away from using `Object.defineProperty` here if you wanted to).

Reply:
Just to clarify, the `Object.definedProperty` doesn't have anything to do with TypeScript. It's part of the original code for this hook: https://github.com/apollographql/apollo-client-devtools/blob/master/src/backend/hook.js#L67

Reply:
Got it, thanks @jcreighton - `Object.defineProperty` can be used to work around type errors, so this was just a heads up in case that was the intent here.